src/pyggdrasil/tree_inference/_simulate.py

```diff
@@ -587,15 +587,27 @@ def adjacency_to_root_dfs(
     return root
 
 
+class CellSimulationParams(TypedDict):
+    """Cell Simulation Parameters."""
+
+    n_cells: int
+    n_mutations: int
+    fpr: float
+    fnr: float
+    na_rate: float
+    observe_homozygous: bool
+    strategy: str
+
+
 def gen_sim_data(
-    params: dict,
+    params: CellSimulationParams,
     rng: interface.JAXRandomKey,
 ) -> dict:
     """
     Generates cell mutation matrix for one tree.
 
     Args:
-        params: input parameters from parser
+        params: TypedDict from parser of cell_simulation.py
```

Comment:
Reading this code again - now with the knowledge about `TypedDict`s - I felt the urge to do at least a little bit of type checking/enforcing parameters here.

Reply:
For validation I really recommend [Pydantic](https://docs.pydantic.dev/latest/usage/models/), it's really excellent.

Reply:
Added a Pydantic model now :)